src/models/audio_processor.cpp

```diff
@@ -0,0 +1,103 @@
+              input_features_value->GetTensorMutableData<float>());
+  } else {
+    auto input_features_fp32 = OrtValue::CreateTensor<float>(
+        allocator.GetInfo(),
+        std::span<float>(const_cast<float*>(mel_data), input_features_value->GetTensorTypeAndShapeInfo()->GetElementCount()),
+        shape_span);
+    ConvertFp32ToFp16(allocator, *input_features_fp32, input_features_value, DeviceType::CPU, nullptr);
+  }
+
+  return input_features_value;
+}
+
+}  // namespace
+
+std::unique_ptr<Audios> LoadAudios(const std::span<const char* const>& audio_paths) {
+  for (const char* audio_path : audio_paths) {
+    if (!fs::path(audio_path).exists()) {
+      throw std::runtime_error("Audio path does not exist: " + std::string(audio_path));
+    }
+  }
+  ort_extensions::OrtxObjectPtr<OrtxRawAudios> audios;
+  CheckResult(OrtxLoadAudios(ort_extensions::ptr(audios), audio_paths.data(), audio_paths.size()));
+
+  return std::make_unique<Audios>(std::move(audios), audio_paths.size());
+}
+
+AudioProcessor::AudioProcessor(Config& config, const SessionInfo& session_info)
+    : input_features_type_{session_info.GetInputDataType(config.model.encoder_decoder_init.inputs.input_features)} {
+  const std::string default_processor_file_name = "audio_processor_config.json";
+  auto processor_config = (config.config_path / fs::path(default_processor_file_name)).string();
+  processor_ = ort_extensions::OrtxObjectPtr<OrtxFeatureExtractor>(OrtxCreateSpeechFeatureExtractor, processor_config.c_str());
+
+  config.AddMapping(std::string(Config::Defaults::InputFeaturesName), config.model.encoder_decoder_init.inputs.input_features);
+}
+
+std::unique_ptr<NamedTensors> AudioProcessor::Process(const Tokenizer& tokenizer, const Audios* audios,
+                                                      const std::string& language, const std::string& task,
+                                                      int32_t no_timestamps) const {
+  if (!audios || !audios->audios_) {
+    throw std::runtime_error("No audios provided to process.");
+  }
+
+  Ort::Allocator& allocator{Ort::Allocator::GetWithDefaultOptions()};
+  auto named_tensors = std::make_unique<NamedTensors>();
+
+  ort_extensions::OrtxObjectPtr<OrtxTensorResult> result;
+  CheckResult(OrtxSpeechLogMel(processor_.get(), audios->audios_.get(), ort_extensions::ptr(result)));
+
+  ort_extensions::OrtxObjectPtr<OrtxTensor> mel;
+  CheckResult(OrtxTensorResultGetAt(result.get(), 0, ort_extensions::ptr(mel)));
+
+  named_tensors->emplace(std::string(Config::Defaults::InputFeaturesName),
+                         std::make_shared<Tensor>(ProcessMel(mel, input_features_type_, allocator)));
+
+  constexpr auto start_of_transcript = "<|startoftranscript|>";
+  const int32_t start_of_transcript_token_id = tokenizer.TokenToTokenId(start_of_transcript);
+  const auto prompt_token_ids = tokenizer.GetDecoderPromptIds(audios->num_audios_, language, task, no_timestamps);
+
```

Comment:
can we treat those as an prompt of the whisper model and allow user to compose this by themselves?

Reply:
and Whispler allows user to provide 'prev' tokens to help get better transcription. It can also be part of the prompt of the whipser model

Reply:
The problem with doing this approach is that then we need to detect `<|startoftranscript|>` in the user prompt. The `<|startoftranscript|>` may not always map to a single token when encoding with the tokenizer.

We can handle this by detecting model type here or by asking users to provide the token ids instead of a string prompt.

But I think we need to still add the ability for users to supply previously generated tokens which is not available right now. I'll add this functionality.